src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -366,13 +412,10 @@ public void triggerFleetStatusUpdateAtKernelLaunch() {
      */
     public void updateFleetStatusUpdateForAllComponents(Trigger trigger) {
         Set<GreengrassService> greengrassServiceSet = new HashSet<>();
-        AtomicReference<OverallStatus> overAllStatus = new AtomicReference<>(OverallStatus.HEALTHY);
-
-        // Get all running services from the Nucleus to update the fleet status.
-        this.kernel.orderedDependencies().forEach(greengrassService -> {
-            greengrassServiceSet.add(greengrassService);
-            overAllStatus.set(getOverallStatusBasedOnServiceState(overAllStatus.get(), greengrassService));
-        });
+        AtomicReference<OverallStatus> overAllStatus = updateDeviceOverallStatus();
+        for (GreengrassService service : serviceFssTracksMap.keySet()) {
```

Comment:
i believe you can replace this entire loop with a bulk set constructor call like
`Set<GreengrassService> greengrassServiceSet = new HashSet<>(serviceFssTracksMap.keySet());`

Reply:
Updated